client/my-sites/controller.js

```diff
@@ -129,50 +126,40 @@ function renderNoVisibleSites( context ) {
 
 	removeSidebar( context );
 
-	renderWithReduxStore(
-		React.createElement( EmptyContentComponent, {
-			title: i18n.translate(
-				'You have %(hidden)d hidden WordPress site.',
-				'You have %(hidden)d hidden WordPress sites.',
-				{
-					count: hiddenSites,
-					args: { hidden: hiddenSites },
-				}
-			),
+	context.primary = React.createElement( EmptyContentComponent, {
+		title: i18n.translate(
+			'You have %(hidden)d hidden WordPress site.',
+			'You have %(hidden)d hidden WordPress sites.',
+			{
+				count: hiddenSites,
+				args: { hidden: hiddenSites },
+			}
+		),
 
-			line: i18n.translate(
-				'To manage it here, set it to visible.',
-				'To manage them here, set them to visible.',
-				{
-					count: hiddenSites,
-				}
-			),
-
-			action: i18n.translate( 'Change Visibility' ),
-			actionURL: '//dashboard.wordpress.com/wp-admin/index.php?page=my-blogs',
-			secondaryAction: i18n.translate( 'Create New Site' ),
-			secondaryActionURL: `${ signup_url }?ref=calypso-nosites`,
-		} ),
-		document.getElementById( 'primary' ),
-		context.store
-	);
+		line: i18n.translate(
+			'To manage it here, set it to visible.',
+			'To manage them here, set them to visible.',
+			{
+				count: hiddenSites,
+			}
+		),
+
+		action: i18n.translate( 'Change Visibility' ),
+		actionURL: '//dashboard.wordpress.com/wp-admin/index.php?page=my-blogs',
+		secondaryAction: i18n.translate( 'Create New Site' ),
+		secondaryActionURL: `${ signup_url }?ref=calypso-nosites`,
+	} );
+
+	makeLayout( context, noop );
+	clientRender( context );
 }
 
 function renderSelectedSiteIsDomainOnly( reactContext, selectedSite ) {
 	const DomainOnly = require( 'my-sites/domains/domain-management/list/domain-only' );
-	const { store: reduxStore } = reactContext;
 
-	renderWithReduxStore(
-		<DomainOnly siteId={ selectedSite.ID } hasNotice={ false } />,
-		document.getElementById( 'primary' ),
-		reduxStore
-	);
+	reactContext.primary = <DomainOnly siteId={ selectedSite.ID } hasNotice={ false } />;
 
-	renderWithReduxStore(
-		createNavigation( reactContext ),
-		document.getElementById( 'secondary' ),
-		reduxStore
-	);
+	reactContext.secondary = createNavigation( reactContext );
```

Comment:
You're missing:
```
makeLayout( context, noop );
clientRender( context );
```

This should solve what @klimeryk was seeing.

Reply:
Thanks, yep you're right. This `siteSelection()` is quite a beast!